controllers/rating.js

```diff
@@ -0,0 +1,43 @@
+import models from '../models';
+
+const { Article } = models;
+
+export default {
+  async rateArticle(req, res) {
+    const { params: { slug }, decoded: { id } } = req;
+    let { rateValue } = req.body;
+    rateValue = Number(rateValue);
+
+    const article = await Article.findOne({ where: { slug } });
+    const { rating } = article;
+
+    if (!rating) {
+      await article.update({
+        rating: { [id]: rateValue },
+        aveRating: rateValue
+      });
+    }
+
+    if (rating) {
+      rating[id] = rateValue;
+      await article.update({
+        rating
+      });
+
+      const ratingArray = Object.values(rating);
+      const sum = ratingArray.reduce((a, b) => a + b);
+      const aveRating = Math.round((sum / ratingArray.length) * 10) / 10;
+
+      await article.update({
```

Comment:
Could we merge the two updates here too?

Reply:
I left this out because the average rating needs the first update to be done before it so that it can use the new value.

Reply:
Can this be done any other way?

Reply:
Yes, I get what you are saying. I've made the change.
Thanks.